system/lib/libc/musl/src/internal/libc.h

```diff
@@ -5,31 +5,35 @@
 #include <stdio.h>
 #include <limits.h>
 
+struct __locale_map;
+
+struct __locale_struct {
+	const struct __locale_map *volatile cat[6];
+};
+
+struct tls_module {
+	struct tls_module *next;
+	void *image;
+	size_t len, size, align, offset;
+};
+
 struct __libc {
-	void *main_thread;
+	int can_do_threads;
```

Comment:
This looks interesting. Perhaps we want to make sure that this gets initialized appropriately in `-s USE_PTHREADS=0/1/2` builds.